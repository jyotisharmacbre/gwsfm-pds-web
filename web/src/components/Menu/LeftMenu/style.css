
/* Fixed sidenav, full height */
.sidenav {
    width: 100%;
    background-color: #393939;
    overflow-x: hidden;
    padding-top: 20px;
  }
  
  /* Style the sidenav links and the dropdown button */
  .sidenav a, .dropdown-btn {
    padding: 10px 8px 10px 16px;
    text-decoration: none;
    font-size: 15px;
    color: white;
    display: block;
    border: none;
    background: none;
    width: 100%;
    text-align: left;
    cursor: pointer;
    outline: none;
  }
  
  /* On mouse-over */
  .sidenav a:hover, .dropdown-btn:hover {
    color: #f1f1f1;
  }
  
  /* Main content */
  .main {
    margin-left: 200px; /* Same as the width of the sidenav */
    font-size: 15px; /* Increased text to enable scrolling */
    padding: 0px 10px;
  }
  
  /* Add an active class to the active dropdown button */
  .active {
    border-left: 6px solid #6DD400;
    color: white;
    font-weight: 600;
    background-color: black;
  }

  .active-sub {
    color: #6DD400 !important;
  }
  
  /* Dropdown container (hidden by default). Optional: add a lighter background color and some left padding to change the design of the dropdown content */
  .dropdown-container {
    display: none;
    background-color: #393939;
    padding-left: 8px;
  }
  
  /* Optional: Style the caret down icon */
  .fa-caret-down {
    float: right;
    padding-right: 8px;
  }
  
  /* Some media queries for responsiveness */
  @media screen and (max-height: 450px) {
    .sidenav {padding-top: 15px;}
    .sidenav a {font-size: 18px;}
  }

  .showmenu {
      display: block !important;
  }

  .hidemenu{
    display: none ;
  }