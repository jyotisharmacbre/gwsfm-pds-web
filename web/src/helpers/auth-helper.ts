// @ts-ignore
import authentication from '@kdpw/msal-b2c-react';
import jwtDecode from 'jwt-decode';
import { DecodedToken } from '../models/decodedToken';

export const getDisplayName = () => {
  const decoded: DecodedToken = decodeToken();
  return decoded.name;
};

export const getFirstName = () => {
  const decoded: DecodedToken = decodeToken();
  return decoded.given_name;
};

export const getDisplayEmail = () => {
  const decoded: DecodedToken = decodeToken();
  return decoded.email ?
    decoded.email :
    decoded.emails && decoded.emails.length > 0 ?
      decoded.emails[0] :
      '';
};

export const logOut = () => {
  authentication.signOut();
}

function decodeToken() {
  const token = authentication.getAccessToken();
  const decoded: DecodedToken = jwtDecode(token);
  return decoded;
}

