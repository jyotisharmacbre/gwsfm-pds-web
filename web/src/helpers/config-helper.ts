const appConfig = () => {
    const config: any = (window as any).Config;
    if(config && config.REACT_APP_AUTH_INSTANCE==="#{AUTH_INSTANCE}#"){
      return {
        ...process.env
      }
    }
    return {
      ...config
    }
  };
  
  export default appConfig;