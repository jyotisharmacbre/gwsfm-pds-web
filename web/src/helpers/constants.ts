export const selectionButtons = [
  {
    title: 'BUTTON_YES',
    value: true
  },
  {
    title: 'BUTTON_NO',
    value: false
  }
];

export enum Notify {
  none,
  success,
  error
}
