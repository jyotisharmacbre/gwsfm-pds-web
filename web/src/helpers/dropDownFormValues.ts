export const projectStatusData: object[] = [
  { label: '1', value: '1' },
  { label: '2', value: '2' },
  { label: '3', value: '3' }
];

export const engagementData: object[] = [
  { label: 'Type 1', value: '1' },
  { label: 'Type 2', value: '2' },
  { label: 'Type 3', value: '3' }
];

export const typesOfEngagement: object[] = [
  {
    label: '---Please choose an option---',
    value: ''
  },
  {
    label: 'Fixed Price',
    value: 'Fixed Price'
  },
  {
    label: 'Time and Materials',
    value: 'Time and Materials'
  },
  {
    label: 'Risk and Reward',
    value: 'Risk and Reward'
  },
  {
    label: 'Cost Plus',
    value: 'Cost Plus'
  }
];

export const contractType: object[] = [
  {
    label: '---Please choose an option---',
    value: ''
  },
  {
    label: 'CT 1',
    value: 'CT 1'
  },
  {
    label: 'CT 2',
    value: 'CT 2'
  }
];

export const currency: object[] = [
  {
    label: '---Please choose an option---',
    value: ''
  },
  {
    label: 'AED (UAE Dirham)',
    value: 'AED'
  },
  {
    label: 'AFN (Afghani)',
    value: 'AFN'
  },
  {
    label: 'ALL (Lek)',
    value: 'ALL'
  },
  {
    label: 'AMD (Dram)',
    value: 'AMD'
  },
  {
    label: 'ANG (Dutch Antilles Guilder)',
    value: 'ANG'
  },
  {
    label: 'AOA (Kwanza)',
    value: 'AOA'
  },
  {
    label: 'ARS (Peso)',
    value: 'ARS'
  }
];

export const projectStatus: object[] = [
  {
    label: '---Please choose an option---',
    value: ''
  },
  {
    label: 'Initial Customer Inquiry',
    value: 'Initial Customer Inquiry'
  },
  {
    label: 'J&A',
    value: 'J&A'
  },
  {
    label: 'Archived',
    value: 'Archived'
  },
  {
    label: 'Lost',
    value: 'Lost'
  },
  {
    label: 'Order Received',
    value: 'Order Received'
  },
  {
    label: 'CPP',
    value: 'CPP'
  },
  {
    label: 'Complete',
    value: 'Complete'
  }
];

export const assetsWorkedOn: object[] = [
  {
    label: '---Please choose an option---',
    value: ''
  },
  {
    label: 'AS 1',
    value: 'AS 1'
  },
  {
    label: 'AS 2',
    value: 'AS 2'
  }
];

export const enquiryTypeData: object[] = [
  { label: 'Written', value: '1' },
  { label: 'Verbal', value: '2' },
  { label: 'Full tender enquiry with specifications & orgs', value: '3' },
  { label: 'Other', value: '4' }
];
