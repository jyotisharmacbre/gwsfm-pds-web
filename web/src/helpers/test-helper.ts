import checkPropTypes from 'check-prop-types';

export const findByTestAtrr = (component: any, attr: string) => {
  const wrapper = component.find(`[data-test='${attr}']`);
  return wrapper;
};

export const checkProps = (component: any, expectedProps: any) => {
  const propsErr = checkPropTypes(
    component.propTypes,
    expectedProps,
    'props',
    component.name
  );
  return propsErr;
};
