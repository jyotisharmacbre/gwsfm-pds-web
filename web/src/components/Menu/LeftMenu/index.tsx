import React from 'react';
import {useHistory } from 'react-router-dom';
import cbre_icon from '../../images/logo-black.png';
import { IState } from '../../../store/state';
import { connect } from 'react-redux';
import { isValidGUID } from '../../../helpers/utility-helper';
import { FormattedMessage, injectIntl } from 'react-intl';
import ProjectStatus from '../../../enums/ProjectStatus';
import { isDirty, reset } from 'redux-form';
import { confirmAlert } from '../../Popup/CustomModalPopup';
import IReactIntl from '../../../Translations/IReactIntl';

interface IMapStateToProps {
	projectId: string;
	status: number;
	isProjectFormDirty:boolean,
	isProjectOverviewFormDirty:boolean,
	isPreliminaryFormDirty:boolean,
	isSubContractorFormDirty:boolean,
	isDiscountFormDirty:boolean,
	intl:any
}
interface IMapDispatchToProps {
	resetProjectFormState:()=>void;
	resetProjectOverviewFormState:()=>void;
	resetPreliminaryFormState:()=>void;
	resetSubContractorFormState:()=>void;
	resetDiscountFormState:()=>void;
}
const LeftMenu: React.FC<IMapStateToProps&IMapDispatchToProps&IReactIntl> = (props) => {
	let urlProjectId: string = '';
	let activeClass: string = '';
	let links: Array<string> = [
		'project',
		'projectoverview',
		'justificationauthorisation',
		'preliminaries',
		'subcontractor',
		'discounts',
		'reviewsubmit',
		'reviewapprove'
	];
	let history = useHistory();
	
	const getGUID = () => {
		history.location.pathname.split('/').forEach((data) => {
			if (isValidGUID(data)) {
				urlProjectId = data;
				return;
			}
		});
		return urlProjectId;
	};
	const getUrlPathName = () => {
		history.location.pathname.split('/').forEach((data) => {
			if (links.includes(data.toLowerCase())) {
				activeClass = data;
				return;
			}
		});
		return activeClass;
	};
	urlProjectId = props.projectId ? props.projectId : getGUID();
	activeClass = activeClass ? activeClass : getUrlPathName().toLowerCase();
	let isDisable: boolean = urlProjectId && urlProjectId != 'undefined' ? true : false;
	
	const disableEnableMenu = (value: string) => {
		return activeClass == value ? 'active' : '';
	};
	const disableEnableSubActiveClass = (value: string) => {
		return activeClass == value ? 'subactive' : '';
	};
	const isFormDirty=(componentName:string,projectId:string)=>{
if( props.isProjectFormDirty||
	props.isProjectOverviewFormDirty||
	props.isPreliminaryFormDirty||
	props.isSubContractorFormDirty||
	props.isDiscountFormDirty)
	{
		confirmAlert({
			intl: props.intl,
			titleKey: 'TITLE_CONFIRMATION',
			contentKey: 'MESSAGE_DIRTY_CHECK',
			handleConfirm: () => redirectionToComponent(componentName,projectId)
		})
	}
	else
	{
		redirectionToComponent(componentName,projectId);
	}
	}
    const redirectionToComponent=(componentName:string,projectId:string)=>{
		if(props.isDiscountFormDirty){props.resetDiscountFormState();}
		if(props.isProjectFormDirty){props.resetProjectFormState();}
		if(props.isProjectOverviewFormDirty){props.resetProjectOverviewFormState();}
		if(props.isPreliminaryFormDirty){props.resetPreliminaryFormState();}
		if(props.isSubContractorFormDirty){props.resetSubContractorFormState();}
		history.push(`/${componentName}/${projectId}`);
	}
	return (
		<nav id="sidebar">
			<div className="sidebar-header">
				<div id="sm_none" className="logo">
					<a className="cursorPntr"
						data-test=""
						onClick={()=>isFormDirty('','')}
					>
						<img src={cbre_icon} alt="CBRE PDS" onClick={()=>isFormDirty('','')} />
					</a>
				</div>
				<div className="cross-menu">
					MENU
					<button type="button" id="cross-inner" className="cross-sidebar">
						<span />
						<span />
						<span />
					</button>
				</div>
			</div>

			<ul id="homeMenu" className="list-unstyled components">
				<li className={disableEnableMenu('project')}>
					<a className="cursorPntr"
						data-test="ProjectLink" onClick={()=>isFormDirty('Project',urlProjectId)}	
					>
						<FormattedMessage id="HEADING_CUSTOMER_ENQUIRY" />
					</a>
				</li>
				<li
					data-test="ProjectOverviewLink"
					className={isDisable ? disableEnableMenu('projectoverview') : 'link_disabled'}
				>
					<a className="cursorPntr"
						data-test="ProjectOverviewPath"
						onClick={()=>isFormDirty('ProjectOverview',urlProjectId)}
					>
						<FormattedMessage id="MENU_PROJECT_OVERVIEW" />
					</a>
				</li>
				<li
					className={
						isDisable ? (
							disableEnableMenu('justificationauthorisation') ||
							disableEnableMenu('preliminaries') ||
							disableEnableMenu('subcontractor') ||
							disableEnableMenu('discounts')
						) : (
							'link_disabled'
						)
					}
				>
					<a  
						data-target="#homeSubmenu"
						data-toggle="collapse"
						aria-expanded="true"
						onClick={()=>isFormDirty('JustificationAuthorisation',urlProjectId)}
						className="dropdown-toggle collapsed cursorPntr"
					>
						<FormattedMessage id="MENU_JUSTIFICATION" />
					</a>
					<ul className="collapse list-unstyled show" id="homeSubmenu">
						<li className={disableEnableSubActiveClass('preliminaries')}>
							<a className="cursorPntr" onClick={()=>isFormDirty('preliminaries',urlProjectId)}>
								<FormattedMessage id="MENU_PRELIMINARIES" />
							</a>
						</li>
						<li className={disableEnableSubActiveClass('subcontractor')}>
							<a className="cursorPntr" onClick={()=>isFormDirty('Subcontractor',urlProjectId)} >
								<FormattedMessage id="MENU_SUBCONTRACTORS" />
							</a>
						</li>
						<li className={disableEnableSubActiveClass('discounts')}>
							<a className="cursorPntr" onClick={()=>isFormDirty('Discounts',urlProjectId)} >
								<FormattedMessage id="MENU_DISCOUNTS" />{' '}
							</a>
						</li>
					</ul>
				</li>
				<li className={isDisable ? disableEnableMenu('reviewsubmit') : 'link_disabled'}>
					<a className="cursorPntr"
					 onClick={()=>isFormDirty('ReviewSubmit',urlProjectId)}
					>
						<FormattedMessage id="MENU_REVIEW_SUBMIT" />
					</a>
				</li>
				{props.status == ProjectStatus.InReview && (
					<li
						data-test="review-approve"
						className={isDisable ? disableEnableMenu('reviewapprove') : 'link_disabled'}
					>
						<a className="cursorPntr"
						onClick={()=>isFormDirty('ReviewApprove',urlProjectId)}
						>
							<FormattedMessage id="MENU_REVIEW_APPROVE" />
						</a>
					</li>
				)}
				<li className={isDisable ? '' : 'link_disabled'}>
					<a className="cursorPntr" onClick={()=>isFormDirty('','')}>
						<FormattedMessage id="MENU_LOGOUT" />
					</a>
				</li>
			</ul>
		</nav>
	);
};

const mapStateToProps = (state: IState) => {
	return {
		isProjectFormDirty:isDirty("ProjectForm")(state),
		isProjectOverviewFormDirty:isDirty("projectOverviewForm")(state),
		isPreliminaryFormDirty:isDirty("PreliminaryForm")(state),
		isSubContractorFormDirty:isDirty("subContractorForm")(state),
		isDiscountFormDirty:isDirty("DiscountForm")(state),
		projectId: state.project.form.projectId,
		status: state.project.form.status
	};
};
const mapDispatchToProps = (dispatch) => {
	return {
		resetProjectFormState:()=>dispatch(reset("ProjectForm")),
		resetProjectOverviewFormState:()=>dispatch(reset("projectOverviewForm")),
		resetPreliminaryFormState:()=>dispatch(reset("PreliminaryForm")),
		resetSubContractorFormState:()=>dispatch(reset("subContractorForm")),
		resetDiscountFormState:()=>dispatch(reset("DiscountForm"))
	};
};
export default injectIntl(connect(mapStateToProps,mapDispatchToProps)(LeftMenu));
