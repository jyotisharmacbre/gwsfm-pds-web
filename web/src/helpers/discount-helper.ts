import DiscountType from '../enums/DiscountType';
import { restrictMinusAndAllowDecimalForMaxRangeHundred, restrictMinusAndAllowDecimal } from '../helpers/utility-helper';

const normalizeClientDiscount = (discountType: DiscountType, discountValue) => {
    let updatedDiscount;
    if (discountType === DiscountType.Percent) {
        updatedDiscount = restrictMinusAndAllowDecimalForMaxRangeHundred(discountValue);
    }
    else updatedDiscount = restrictMinusAndAllowDecimal(discountValue);
    return updatedDiscount;
}

export default normalizeClientDiscount;