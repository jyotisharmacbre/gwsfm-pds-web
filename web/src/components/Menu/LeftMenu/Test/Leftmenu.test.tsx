
 
import React from 'react';
import { mount} from 'enzyme';
import { Provider } from 'react-redux';
import {findByTestAtrr} from "../../../../helpers/test-helper"
import { IntlProvider } from 'react-intl';
import translations from '../../../../Translations/translation';
import configureStore from 'redux-mock-store';
import LeftMenu from '../index';
import { leftMenuProjectData } from './LeftMenuTestData';
import {BrowserRouter} from 'react-router-dom';

let props: any = {
    projectId : ""
  };
  describe('Left Menu Renders', () => {
    const mockStore = configureStore([]);
    let store;
    let wrapper;  
    beforeEach(() => {
        store = mockStore({
            project:leftMenuProjectData
          });
          jest.mock('react-router-dom', () => ({
            useHistory: jest.fn().mockReturnValue({  length: 13,
              push: jest.fn(),
              block: jest.fn(),
              createHref: jest.fn(),
              go: jest.fn(),
              goBack: jest.fn(),
              goForward: jest.fn(),
              liten: jest.fn(),
              replace: jest.fn(),
              action: "REPLACE",
              location: null }),
          }));
          props.projectId="";
          leftMenuProjectData.form.projectId="";
                wrapper = mount(
        <Provider store={store}>
        <IntlProvider locale="en" messages={translations['en'].messages}>
        <BrowserRouter>
        <LeftMenu {...props}/>
        </BrowserRouter>
        </IntlProvider>
        </Provider>
        
      );
    });  
  it('Defines the component', () => {
   
    expect(wrapper).toBeDefined();
  }); 
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  }); 
  it('should disable the link if project id is not found', () => {
    const container = findByTestAtrr(wrapper,'ProjectOverviewLink').first();
    expect(container.hasClass("link_disabled")).toBe(true);
  });
  it('should enable and pass project id with url,if project id is there', () => {
    leftMenuProjectData.form.projectId="309ccd02-38ab-4643-1165-08d77e00a6ce";
    wrapper = mount(
  <Provider store={store}>
  <IntlProvider locale="en" messages={translations['en'].messages}>
  <BrowserRouter>
  <LeftMenu {...props}/>
  </BrowserRouter>
  </IntlProvider>
  </Provider>
          )
    const container = findByTestAtrr(wrapper,'ProjectOverviewLink').first();
    const link = findByTestAtrr(wrapper,'ProjectOverviewPath').first();
    expect(container.hasClass("link_disabled")).toBe(false);
   expect(link.getDOMNode().href).toContain("309ccd02-38ab-4643-1165-08d77e00a6ce");
  });
  it('should disable the link if project is not in GUID format inside url', () => {
    window.location.href="http://localhost/projectOverview/309ccd02-38ab-4643-1165-08d7";
    const container = findByTestAtrr(wrapper,'ProjectOverviewLink').first();
    const link = findByTestAtrr(wrapper,'ProjectOverviewPath').first();
    expect(link.getDOMNode().href).not.toContain("309ccd02-38ab-4643-1165-08d7");
    expect(container.hasClass("link_disabled")).toBe(true);
  });
});