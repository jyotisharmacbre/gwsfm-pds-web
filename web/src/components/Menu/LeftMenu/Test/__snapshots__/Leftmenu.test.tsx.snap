// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Left Menu Renders should match the snapshot 1`] = `ReactWrapper {}`;
